---
title: "Microbiome analysis of Germany and Sweden"
author: "Talia_Karasov"
date: "5/3/2019"
output: html_document
---
```{r global_options, include=FALSE}
knitr::opts_chunk$set(fig.width = 12, fig.height = 8, fig.path ='Figs/',
                      echo = FALSE, warning = FALSE, message = FALSE, cache = TRUE)
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(reshape2)
library('dplyr')
library('tidyr')
library(ggplot2)
library(RColorBrewer)
library(wesanderson)
library(matrixStats)
require(gridExtra)
library(cowplot)
#library(moments)
library(intrval)
library("Hmisc")
library(ape)
library(cowplot)
library(vegan)
library(extrafont)
library(lme4)
library(ape)
library(ggthemes)
library(knitr)
#font_import()
loadfonts()

source("~/Dropbox/controlled_metagenomics/scripts/keep_used_in_publication/controlled_metagenomics_functions.R")

hue1_12 = c("#ada77c","#4477AA", "#77AADD", "#117777", "#44AAAA", "#77CCCC", "#117744", "#44AA77","#114477","#88CCAA", "#777711", "#AAAA44")
hue13_25 = c("#DDDD77", "#774411", "#AA7744", "#DDAA77", "#771122","#DD7788", "darkgoldenrod1", "#771155", "#AA4488", "#CC99BB","#AA7744", "#AA4455")

hue15_25 = c("#771122","#DD7788", "darkgoldenrod1", "#771155", "#AA4488", "#CC99BB","#AA7744", "#AA4455")

adonis2_sim <-function(select_g_s , populations = populations){
  i = 0 
  keep_pop = matrix(ncol = 7, nrow = 9)
  rownames(keep_pop) = populations
  for(rec in populations){
    i = i+1
    subset = which(g_s_pop_com == rec)
    sampling = sample(subset, 7, replace = FALSE)
    keep_pop[i,] = sampling
  }
  
  g_s_subset = data.frame(select_g_s[unlist(keep_pop),])
  bc_limited = bc_comp[unlist(keep_pop), unlist(keep_pop)]
  adonis_output = adonis2(bc_limited ~ g_s_subset$country / g_s_subset$pop + g_s_subset$load, strata = g_s_subset$country)
  return(adonis_output)
}
```

#Load vs. microbiome composition
This script explores the relationship between load and microbiome for the german and swedish datasets and graphs the microbiomes of these datasets

###Filtering
First thing is to remove the offending columns from the german dataset and then to combine the metagenome files. These tasks are done with the clean_tables script:
```{r, echo = TRUE}
source("~/Dropbox/controlled_metagenomics/scripts/keep_used_in_publication/clean_tables.R")
```


###Analysis
###Permutational multivariate analysis of variance using distance matrices  (WITH BRAY-CURTIS DISTANCES)
###Bacteria
First we will look at those reads classified as bacterial. We will perform principal coordinate analysis on the bray-curtis disimilarity to assess how load relates to similarity of microbiome content.

```{r, echo=FALSE}

concat_name="_bacteria.csv"

g_s = read.csv(paste("~/Dropbox/controlled_metagenomics/data/sweden_germany_combined_nodup", concat_name, sep=''), header = TRUE, row.names = 1, stringsAsFactors = FALSE)

g_s_tot = process_gs(g_s, subset=FALSE)

g_s_otus_filter = g_s_tot[[1]]

g_s_country = g_s_tot[[2]]

g_s_pop = g_s_tot[[3]]

g_s_load = g_s_tot[[4]]

bray_curtis = as.matrix( vegdist(g_s_otus_filter, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE) )
```

We are using adonis to do the multivariate testing.
Here is some advice on nesting:
https://stats.stackexchange.com/questions/188519/adonis-in-vegan-order-of-variables-or-use-of-strata

Difficulty: I have unbalanced designs to which permanova may be sensitive. To test for differences in dispersion I can use the beta dispersion function in vegan

```{r}
#exclude NA
bc_comp = bray_curtis[which(!is.na(g_s_pop)), which(!is.na(g_s_pop))]
g_s_load_com = g_s_load[which(!is.na(g_s_pop))]
g_s_pop_com = g_s_pop[which(!is.na(g_s_pop))]
g_s_country_com = g_s_country[which(!is.na(g_s_pop))]
disp = betadisper(d = as.dist(bc_comp), group = as.factor(g_s_country_com))
permutest(disp)
```

That's not good. The dispersion does seem to be significantly different between groups. 

I guess we need to bootstrap the NextMet samples
```{r}
table(g_s_pop_com)
table(g_s_country_com)
german = table(g_s_country_com)[1]
sweden = table(g_s_country_com)[2]
german_list = which(g_s_country_com == "germany")
swedish_list = which(g_s_country_com == "Sweden")
```


And now run adonis on each of these subsets
```{r}
#run adonis2 100 times with different subsets from germany
value = matrix(ncol = 2, nrow = 100)
colnames(value) = c("country", "load")

for(i in 1:100){
  german_sub = sample(german_list, size = as.numeric(sweden))
  full = c(german_sub, swedish_list)
  adon = adonis2(bc_comp[full, full] ~ g_s_country_com[full] + g_s_load_com[full])
  value[i,1] = adon$R2[1]
  value[i,2] = adon$R2[2]
}

#pop_mv = adonis2(bc_comp ~ g_s_pop_com / g_s_country_com + g_s_load_com, strata = g_s_pop_com )
#country_mv = adonis2(bc_comp ~ g_s_country_com + g_s_load_com)

```

I think I should also do the pcoa with just a subset of individuals from each population
```{r}
populations = unique(g_s_pop_com)
select_g_s = cbind(g_s_country_com, g_s_pop_com, g_s_load_com)
colnames(select_g_s) = c("country", "pop", "load")

adon_mat = matrix(nrow = 100, ncol = 6)
colnames(adon_mat) = c("R2_country", "R2_load", "R2_pop", "pval_country", "pval_load", "pval_pop")
```

Bootstrap the adonis value 100 times
```{r}

for(i in 1:100){
    z = 0
    keep_pop = matrix(ncol = 8, nrow = 9)
    rownames(keep_pop) = populations
    for(rec in populations){
      z = z + 1
      subset = which(g_s_pop_com == rec)
      sampling = sample(subset, 8, replace = FALSE)
      keep_pop[z,] = sampling
    }
    
  g_s_subset = data.frame(select_g_s[unlist(keep_pop),])
  g_s_subset$load = as.numeric(as.character(g_s_subset$load))
  bc_limited = bc_comp[unlist(keep_pop), unlist(keep_pop)]
  boot = adonis2(bc_limited ~ g_s_subset$country / g_s_subset$pop + g_s_subset$load, strata = g_s_subset$country)
  #boot = adonis2_sim(select_g_s = select_g_s , populations = populations)
  adon_mat[i,1] = boot$R2[1]
  adon_mat[i,2] = boot$R2[2]
  adon_mat[i,3] = boot$R2[3]
  adon_mat[i,4:6] = boot$`Pr(>F)`[1:3]
}

summary(adon_mat)
```

What is the dispersion of this subset?
```{r}
disp =  betadisper(d = as.dist(bc_limited), group = as.factor(g_s_subset$pop))

permutest(disp)

```

###PCoA on bray-curtis of more balanced sample
```{r}
bc_pcoa = pcoa(bc_limited, correction = "lingoes")
perc_explained = bc_pcoa$values[,3]
```

Do PCo1 and PCo2 distinguish between samples from germany vs sweden?
First PCo1:
```{r, echo = FALSE}
pc1 = bc_pcoa$vectors[,1]

german_pc1 = pc1[g_s_subset$country=="germany"]

sweden_pc1 = pc1[g_s_subset$country=="Sweden"]

wilcox.test(german_pc1, sweden_pc1)
#p=0.1486 W=5367
```

Next PC2:
```{r, echo = FALSE}
pc2 = bc_pcoa$vectors[,2]
german_pc2 = pc2[g_s_subset$country=="germany"]
sweden_pc2 = pc2[g_s_subset$country=="Sweden"]
wilcox.test(german_pc2, sweden_pc2)
#p=2.2e-16 W=8645
```

Let's plot an example of subsampled data and PCoA
```{r}
bc_limited = bc_comp[unlist(keep_pop), unlist(keep_pop)]
g_s_pop_limited = g_s_pop_com[unlist(keep_pop)]
g_s_country_limited = g_s_country_com[unlist(keep_pop)]
g_s_load_limited = g_s_load_com[unlist(keep_pop)]

load_pcoa  =  ggplot(data=data.frame(bc_pcoa$vectors), aes(x=Axis.1, y=Axis.2)) + 
  geom_point(aes(color=log10(g_s_load_limited)), cex = 2) +
  scale_color_gradient2(name = "", low = "blue", mid = "white", high = "red") +
  theme_bw() + theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank(), text = element_text(size = 7)) +
  labs(x=NULL, y = NULL) + 
  #xlab(paste(paste("Axis 1:", (100*round(perc_explained[1],3)), sep=" "), "%", sep="")) +
  theme(legend.key.width=unit(1,"cm"), 
        axis.title.y = element_blank(), 
        axis.text.y = element_blank(),
        text = element_text(size = 7),
        legend.text.align = 1, 
        legend.title = element_blank(), 
        legend.position = 'bottom') +
  annotate("text",  x=Inf, y = Inf, label = expression("log"[10]*"(Tot. Bac. Load)"), vjust=1, hjust=1, cex = 5)

country_pcoa = pop_pcoa  = ggplot(data=data.frame(bc_pcoa$vectors), aes(x=Axis.1, y=Axis.2)) + 
  geom_point(aes(color = g_s_country_limited), cex = 2) +
  scale_color_discrete(name="Country") +
  theme_bw() + 
  theme( panel.grid.major = element_blank(), 
         panel.grid.minor = element_blank(), 
         text = element_text(size = 7), 
         legend.key.size = unit(0, 'lines'), 
         axis.title.y = element_blank(),
         legend.title = element_blank(), 
         legend.position = 'bottom',
         legend.spacing.x = unit(0.01, 'cm')) +
 labs(y=NULL, x = NULL) +
 scale_color_calc() +
  labs(y=NULL, x = NULL) +
  guides(colour = guide_legend(override.aes = list(size=4))) +
  annotate("text",  x=Inf, y = Inf, label = "Country", vjust=1, hjust=1, cex = 3)

pop_pcoa  = ggplot(data=data.frame(bc_pcoa$vectors), aes(x=Axis.1, y=Axis.2)) + 
  geom_point(aes(color = g_s_pop_limited), cex = 2) +
  scale_color_discrete(name="pop") +
  scale_color_brewer(palette = "Paired") +
  theme_bw() + 
  theme( panel.grid.major = element_blank(), 
         panel.grid.minor = element_blank(), 
         text = element_text(size = 7), 
         #legend.key.width=unit(1,"cm"), 
         axis.title.y = element_blank(), 
         legend.title = element_blank(), 
         axis.text.y = element_blank(), 
         legend.position = 'bottom',
         legend.spacing.x = unit(.001, 'cm'))  +
  labs(y=NULL, x = NULL) +
  guides(colour = guide_legend(override.aes = list(size=4))) +
  annotate("text",  x=Inf, y = Inf, label = "Population", vjust=1, hjust=1, cex = 3)

gA  =  ggplotGrob(load_pcoa)
gB  =  ggplotGrob(pop_pcoa)
gC  =  ggplotGrob(country_pcoa)
gA$widths  =  gB$widths = gC$widths
gA$heights = gB$heights = gC$heights

grid.arrange(gC, gB, gA, ncol = 3, left = paste(paste("Axis 2:", (100*round(perc_explained[2],3)), sep=" "), "%", sep=""), bottom = paste(paste("Axis 1:", (100*round(perc_explained[1],3)), sep=" "), "%", sep=""))

pdf(paste(paste("~/Dropbox/controlled_metagenomics/results_figures/pcoa_bray_curtis_", concat_name, sep=""), ".pdf", sep=""), family = "ArialMT", useDingbats=FALSE, width = 7.204, height = 2.5)

grid.arrange(gC, gB, gA, ncol = 3, left = paste(paste("Axis 2:", (100*round(perc_explained[2],3)), sep=" "), "%", sep=""), bottom = paste(paste("Axis 1:", (100*round(perc_explained[1],3)), sep=" "), "%", sep=""))

dev.off()
```


What about looking just in Eyach?
```{r}
eyach = which(g_s_pop_com == "Eyach")
g_s_load_com_eyach = g_s_load_com[eyach]
bc_comp_eyach = bc_comp[eyach, eyach]

adonis2(bc_comp_eyach ~ g_s_load_com_eyach)

adonis2(bc_comp ~ g_s_load_com)
```

Which microbe has the highest mean load across samples in a region?
```{r}
find_top_abundance(g_s_otus_filter)
```


What about if I look at the raw read tables?
```{r}


```



What is the relationship between load and PC1?
#```{r}
cor.test(g_s_load, pc1)
```


# ```{r, echo = FALSE}
german_samples = read.table("~/Dropbox/germany_pathogen_collections/sample_data/plate_sample_locations/sample_infoFinal_2018.txt", sep="\t", header=T)

duplicated samples (samples collected or sequenced twice)
keep = german_samples[-which(duplicated(german_samples$uniqueID)),]
no_keep = as.character(german_samples[which(duplicated(german_samples$uniqueID)),]$metagenome_identifier)
#These have too few reads: NextMet175, NextMet193, NextMet194, NextMet55
#These are the controls
#duplicated: NextMet24, NextMet50, NextMet66, NextMet70, NextMet83, NextMet85, NextMet87, NextMet96, NextMet119, NextMet131, NextMet142, NextMet144

keep = keep[which(keep$uniqueID %ni% c("NextMet175", "NextMet193", "NextMet194", "NextMet55")),]
no_keep = c(no_keep, c("NextMet175", "NextMet193", "NextMet194", "NextMet55")) 
```

