---
title: "DC3000 controlled_infections"
author: "Talia_Karasov"
date: "5/6/2019"
output: html_document
---
```{r global_options, include=FALSE}
knitr::opts_chunk$set(fig.width=12, fig.height=8, fig.path='Figs/',
                      echo=FALSE, warning=FALSE, message=FALSE)
```

```{r setup, include=FALSE}
#This script takes the output for HPA and graphs the read progression
#samples that need to be corrected in name before processing:
#D.5.I.5_RunId0091_LaneId1
#control_RunId0091_LaneId1 removed

library(reshape2)
library('dplyr')
library('tidyr')
library(ggplot2)
library(RColorBrewer)
library(wesanderson)
'%!in%' <- function(x,y)!('%in%'(x,y))
mycolors=c("darkblue", "darkgoldenrod1", "darkseagreen", "darkorchid", "darkolivegreen1", "lightskyblue", "firebrick", "khaki2", "brown1", "darkorange1", "cyan1", "royalblue4", "darksalmon", "darkblue","royalblue4", "dodgerblue3", "steelblue1", "lightskyblue", "darkseagreen", "darkgoldenrod1", "darkseagreen", "darkorchid", "darkolivegreen1", "brown1", "darkorange1", "cyan1", "darkgrey")


#meta=read.table("~/work_main/abt6_projects9/metagenomic_controlled/data/processed_reads/hpa_infections/meta_family_corrected_per_plant.csv", sep=",", header=T, row.names = 1)
meta=read.table("/ebio/abt6_projects9/metagenomic_controlled/data/processed_reads/hpa_infections/meta_family_corrected_per_plant.csv", sep=",", header=T, row.names = 1)
meta_hpa=melt(meta[c("Peronosporaceae"),], value.name="load", variable.name="Genotype")
#meta_pseud$Genotype=rownames(meta_pseud)

#exclude control
meta_hpa=meta_hpa[-c(meta_hpa$Genotype=="control_RunId0091_LaneId1"),]

meta_hpa$Day=as.numeric(as.character(gsub("D", "",sapply(strsplit(as.character(meta_hpa$Genotype), ".", fixed=TRUE), `[`, 1))))
meta_hpa$Replicate=sapply(strsplit(as.character(meta_hpa$Genotype), ".", fixed=TRUE), `[`,3)
meta_hpa$Genotype=sapply(strsplit(as.character(meta_hpa$Genotype), ".", fixed=TRUE), `[`,2)
#meta_pseud$Day=sapply(strsplit(sapply(strsplit(meta_pseud$day_ori, ""), `[`, 1), "_R"), `[`,2)
load_plot=meta_hpa%>% group_by(Day, Genotype) %>% summarise(mean = mean(load, na.rm=T), se = sd(load, na.rm=T) / length(load))
```


```{r, echo = FALSE}
plot_hpa_time <- ggplot(load_plot, aes(x=Day, y=mean, group=Genotype, colour=Genotype))+geom_errorbar(aes(ymin=mean-se, ymax=mean+se), width=.1) +
  geom_point()+theme_bw()+geom_line()+theme_bw()+ scale_color_manual(values = wes_palette(n=2, name="GrandBudapest1"))+ylab("Hpa coverage/Plant coverage")

pdf("~/Dropbox/controlled_metagenomics/results_figures/hpa_over_time.pdf",  useDingbats=FALSE, width=6, height=4)
plot_hpa_time + scale_x_continuous(breaks = c(0:11))
dev.off()
#Now for plotting the stacked barplots
#Choose ten most abundant taxa
meta=within(meta, rm("control_RunId0091_LaneId1"))
top10=names(sort(rowSums(meta, na.rm=TRUE), decreasing=TRUE )[1:10])
meta_microbiome=meta[top10,]
rest=colSums(meta[rownames(meta) %!in% top10, ], na.rm=TRUE)
meta_microbiome=rbind(meta_microbiome, rest)
rownames(meta_microbiome)[11]="Rest"
meta_microbiome$Family=rownames(meta_microbiome)
microb_melt=melt(meta_microbiome)
microb_melt$Day=(as.character(gsub("D", "",sapply(strsplit(as.character(microb_melt$variable), ".", fixed=TRUE), `[`, 1))))
microb_melt$Replicate=sapply(strsplit(as.character(microb_melt$variable), ".", fixed=TRUE), `[`,3)
microb_melt$Replicate=sapply(strsplit(as.character(microb_melt$Replicate), "_", fixed=TRUE), `[`,1)
microb_melt$Genotype=sapply(strsplit(as.character(microb_melt$variable), ".", fixed=TRUE), `[`,2)
microb_melt$Day=replace(microb_melt$Day, microb_melt$Day=="11", "664")
microb_melt=microb_melt[order(microb_melt$Day),]
microb_melt=microb_melt[order(microb_melt$Genotype),]
#microb_melt$Day=replace(microb_melt$Day, microb_melt$Day=="11", as.factor("264"))
microb_melt$Gen_Day=paste(microb_melt$Genotype, microb_melt$Day, microb_melt$Replicate, sep=".")

pdf("~/Dropbox/controlled_metagenomics/results_figures/hpa_time_control.pdf")
microb_melt_C=microb_melt[microb_melt$Genotype=="C",]
p2<- ggplot(data=microb_melt_C, aes(x=Gen_Day, y=value, fill=Family))
p2 + geom_bar(aes(), stat="identity", position="stack") +
  scale_fill_manual(values =mycolors) +
  theme(legend.position="bottom", panel.background = element_blank(), axis.line = element_line(colour = "black")) + guides(fill=guide_legend(nrow=5))+ylim(c(0,0.5))+ylab("Coverage Peronospora/Coverage Plant")
dev.off()
 
pdf("~/Dropbox/controlled_metagenomics/results_figures/hpa_time_treated.pdf")
microb_melt_I=microb_melt[microb_melt$Genotype=="I",]
p1<- ggplot(data=microb_melt_I, aes(x=Gen_Day, y=value, fill=Family))
p1 + geom_bar(aes(), stat="identity", position="stack") +
  scale_fill_manual(values = mycolors) +
  theme(legend.position="bottom", panel.background = element_blank(), axis.line = element_line(colour = "black")) + guides(fill=guide_legend(nrow=5))+ylim(c(0,0.5)) +xlab("Day")+ylab("Coverage Peronospora/Coverage Plant")
dev.off()

#make a column for Pseudomonas
per=microb_melt[microb_melt$Family=="Peronosporaceae",]
per$per_val=per$value
per=subset(per,select=-c(Family, value))
meb=merge(microb_melt, per)
meb=meb[which(meb$Genotype!="control"),]
#meb=meb[which(meb$Genotype!="C"),]
lm_Pseud<-lm(log10(value)~log10(per_val), data=meb[meb$Family=="Pseudomonadaceae",])
lm_Per<-lm(log10(value)~log10(per_val), data=meb[meb$Family=="Peronosporaceae",])
lm_Ent<-lm(log10(value)~log10(per_val), data=meb[meb$Family=="Enterobacteriaceae",])
lm_Sphing<-lm(log10(value)~log10(per_val), data=meb[meb$Family=="Sphingomonadaceae",])
lm_Com<-lm(log10(value)~log10(per_val), data=meb[meb$Family=="Comamonadaceae",])
lm_Xan<-lm(log10(value)~log10(per_val), data=meb[meb$Family=="Xanthomonadaceae",])
lm_Rhiz<-lm(log10(value)~log10(per_val), data=meb[meb$Family=="Rhizobiaceae",])
lm_Morax<-lm(log10(value)~log10(per_val), data=meb[meb$Family=="Moraxellaceae",])
lm_Brady<-lm(log10(value)~log10(per_val), data=meb[meb$Family=="Bradyrhizobiaceae",])
lm_Burk<-lm(log10(value)~log10(per_val), data=meb[meb$Family=="Burkholderiaceae",])
lm_Rest<-lm(log10(value)~log10(per_val), data=meb[meb$Family=="Rest",])


df=data.frame()
p=ggplot(df) +xlim(-3,1)+ylim(-3,1)

family_vec=list( lm_Brady, lm_Burk, lm_Com, lm_Ent, lm_Morax, lm_Per, lm_Pseud, lm_Rest, lm_Rhiz, lm_Sphing, lm_Xan)

for(i in 1:11){
  family=family_vec[i][[1]]
  slope=as.numeric(as.character(family$coefficients[2]))
  intercept=as.numeric(as.character(family$coefficients[1]))
  if(summary(family)$coefficients[,4][2]<0.005 && i!=6){
    p=p+geom_abline(slope=slope, intercept=intercept, color=mycolors[i]) 
    print(i)
  }
}
pdf("~/Dropbox/controlled_metagenomics/results_figures/effect_hpa_other.pdf")
p+panel_border(colour = "Black",size=1)

dev.off()

# Plot all regressions together
plot_regress=function(name_family){
  nam = paste("p_",name_family, sep="")
  #assign(nam, ggplot(data=meb[meb$Family==name_family,], aes(x=per_val, y=value)) +
  #         geom_point() + geom_smooth(method=lm, se=TRUE, col="BLACK") + theme_bw())
  my_subset=meb[meb$Family==name_family,]
  regress = lm(log10(value)~log10(per_val), data=my_subset)
  is_sig=summary(regress)$coefficients[,4][2]
  fit = lm(log10(value)~log10(per_val), data=my_subset)
  rsquared=round(signif(summary(fit)$adj.r.squared, 5), digits=3)
  rsq_nam=paste(" R2",rsquared,sep="=")
  name_family_amend = paste(name_family, rsq_nam, sep=",")
  basic=ggplot(data=my_subset, aes(x=log10(per_val), y=log10(value))) +
    geom_point(cex=0.2) + geom_smooth(method=lm, se=TRUE, col="BLACK")
  sig=FALSE
  if(is_sig<0.005){
    p= basic + theme_bw() +
      annotate("text", label=paste(name_family_amend, "*", sep=""), x=-3, y=-1, cex=2, color="RED") +
      xlim(-6,-.5) + ylim(-5,0) +xlab("") + ylab("")#+ xlab(expression(log[10]~("Hpa Load"))) + ylab(expression(log[10]~("Other Load")))
    return(p)}
  else{
    p= basic + theme_bw() +
      annotate("text", label=name_family_amend, x=-3, y=-1, cex=2, color="RED") +
      xlim(-6,-.5) + ylim(-5,0) +xlab("") +ylab("") # + xlab(expression(log[10]~("Hpa Load"))) + ylab(expression(log[10]~("Other Load"))) 
return(p)
  }
}

pdf("/ebio/abt6_projects9/metagenomic_controlled/code/controlled_metagenomics_git/data/hpa_other_taxa_correlations.pdf")
  
families=c("Pseudomonadaceae","Enterobacteriaceae","Sphingomonadaceae","Comamonadaceae","Xanthomonadaceae","Rhizobiaceae","Moraxellaceae","Bradyrhizobiaceae","Burkholderiaceae","Rest")
i=1
plots=rep(0, length(families))
for(fam in families){
  print(fam)
  assign(paste("p_", i, sep=""), plot_regress(fam))
  i=i+1
  }
grid.arrange(p_1,p_2,p_3,p_4,p_5,p_6,p_7,p_8,p_9, bottom=textGrob(expression(log[10]~"(Peron. Cov./Plant Cov.)")), left=textGrob(expression(log[10]~"(Focal Cov./Plant Cov.)"), rot=90))
dev.off()
#PCA on leftover microbiome

```

