---
title: "Figure 2 Diversity"
author: "Talia_Karasov"
date: "5/9/2019"
output: html_document
---

#The relationship between load and diversity and some statistics about differences between populations
The goal of this script is to look at the relationship between load and diversity

```{r global_options, include=FALSE}
knitr::opts_chunk$set(fig.width=12, fig.height=8, fig.path='Figs/',
                      echo=FALSE, warning=FALSE, message=FALSE)
```

```{r setup, include= FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(reshape2)
library('dplyr')
library('tidyr')
library(ggplot2)
library(RColorBrewer)
library(wesanderson)
library(matrixStats)
require(gridExtra)
library(cowplot)
#library(moments)
library(intrval)
library("Hmisc")
library(ape)
library(cowplot)
library(vegan)
library(extrafont)
library(lme4)
#font_import()
loadfonts()
manual_colors=c("darkblue", "darkgoldenrod1", "darkseagreen", "darkorchid", "darkolivegreen1", "lightskyblue", "darkgreen", "deeppink", "khaki2", "firebrick", "brown1", "darkorange1", "cyan1", "royalblue4", "darksalmon", "darkblue", "royalblue4", "dodgerblue3", "steelblue1", "lightskyblue", "darkseagreen", "darkgoldenrod1", "darkseagreen", "darkorchid", "darkolivegreen1", "brown1", "darkorange1", "cyan1", "darkgrey")
source("~/Dropbox/controlled_metagenomics/scripts/keep_used_in_publication/controlled_metagenomics_functions.R")
```

#Functions
```{r, echo = FALSE}

process_gs<-function(g_s, subset=FALSE, classifier=cf, classifier_identity=cfi){
  #function that takes gs
  if(subset==TRUE){
    g_temp<-g_s[which(g_s[[cf]]==cfi),]
  }
  else{
    g_temp<-g_s
  }
  g_s_otus <- g_temp[,which(colnames(g_temp)%ni%c("population", "country", "load"))]
  g_s_load <-g_temp$load
  #[which(apply((g_s_otus), 1, var)!=0)]
  g_s_pop <-g_temp$population
  #[which(apply((g_s_otus), 1, var)!=0)]
  g_s_country <-g_temp$country
  #[which(apply((g_s_otus), 1, var)!=0)]
  #g_s_otus <-g_s[,which(apply((g_s), 1, var)!=0)]
  #first samples that are missing
  g_s_otus <-g_s_otus
  #[which(apply((g_s_otus), 1, var)!=0),]
  #Next taxa that are absent
  g_s_otus <-g_s_otus
  #[which(apply(t(g_s_otus), 2, var)!=0),]
  #remove families that are 
  keep_0.001 <- which(colSums(g_s_otus)/dim(g_s_otus)[1]>0.00001)
  g_s_otus_filter <- g_s_otus[,keep_0.001]
  #now make compositional
  g_s_otus_filter <- g_s_otus_filter/rowSums(g_s_otus)
  together<-list(g_s_otus_filter, g_s_country, g_s_pop, g_s_load)
  return(together)
}
```

Let's look at the relationship in Germany:

First load the bacteria
```{r, echo = FALSE}
concat_name="_bacteria.csv"
g_s_bacteria=read.csv(paste("~/Dropbox/controlled_metagenomics/results_figures/sweden_germany_combined", concat_name, sep=''), header = TRUE, row.names = 1, stringsAsFactors = FALSE)
g_s_tot_bacteria = process_gs(g_s_bacteria, subset=FALSE)

concat_name="_oomycete.csv"
g_s_oomycete=read.csv(paste("~/Dropbox/controlled_metagenomics/results_figures/sweden_germany_combined", concat_name, sep=''), header = TRUE, row.names = 1, stringsAsFactors = FALSE)
g_s_tot_oomycete = process_gs(g_s_oomycete, subset=FALSE)

concat_name="_fungi.csv"
g_s_fungi=read.csv(paste("~/Dropbox/controlled_metagenomics/results_figures/sweden_germany_combined", concat_name, sep=''), header = TRUE, row.names = 1, stringsAsFactors = FALSE)
g_s_tot_fungi = process_gs(g_s_fungi, subset=FALSE)
```

and find samples that have variance in all three:
```{r}
g_s_otus_filter_bacteria = g_s_tot_bacteria[[1]]

g_s_country = g_s_tot_bacteria[[2]]

g_s_pop_bacteria = g_s_tot_bacteria[[3]]

g_s_load_bacteria = g_s_tot_bacteria[[4]]

```

##Oomycetes
```{r, echo=FALSE}
concat_name="_oomycete.csv"

g_s_oom=read.csv(paste("~/Dropbox/controlled_metagenomics/results_figures/sweden_germany_combined", concat_name, sep=''), header = TRUE, row.names = 1, stringsAsFactors = FALSE)

g_s_tot_oom = process_gs(g_s, subset=FALSE)

g_s_otus_filter_oom = g_s_tot[[1]]

g_s_country_oom = g_s_tot[[2]]

g_s_pop_oom = g_s_tot[[3]]

g_s_load_oom = g_s_tot[[4]]

max_tax_oom=apply(g_s_otus_filter_oom, 1, max)
max_tax_names_oom = colnames(g_s_otus_filter_oom)[as.integer(apply(g_s_otus_filter_oom, 1, which.max))]

```

##And fungi:
```{r, echo = FALSE}

g_s_otus_filter_fungi = g_s_tot_fungi[[1]]

g_s_country = g_s_tot_fungi[[2]]

g_s_pop_fungi = g_s_tot_fungi[[3]]

g_s_load_fungi = g_s_tot_fungi[[4]]
```


###How is load associated with Shannon Diversity

####Calculate diversity for bacteria
```{r, echo=FALSE}
shannon_div <- vegan::diversity(g_s_otus_filter_bacteria, index = "shannon")
simpson_div <- vegan::diversity(g_s_otus_filter_bacteria, index="simpson")
J <- shannon_div/log(specnumber(g_s_otus_filter_bacteria))
max_tax_bac=apply(g_s_otus_filter_bacteria, 1, max)
max_tax_names = colnames(g_s_otus_filter_bacteria)[as.integer(apply(g_s_otus_filter_bacteria, 1, which.max))]
div_tog_bac = bind_cols(load_bac = g_s_load_bacteria, load_oom = g_s_load_oomycete, load_fungi = g_s_load_fungi, shannon = shannon_div, simpson = simpson_div, J=J, max_tax = max_tax_bac)
```

####Calculate diversity for Oomycetes
```{r}
shannon_div_oom <- vegan::diversity(g_s_otus_filter_oom, index = "shannon")
simpson_div_oom <- vegan::diversity(g_s_otus_filter_oom, index="simpson")
J_oom <- shannon_div_oom/log(specnumber(g_s_otus_filter_oom))
max_tax_oom = apply(g_s_otus_filter_oom, 1, max)
max_tax_names_oom = colnames(g_s_otus_filter_oom)[apply(g_s_otus_filter_oom, 1, which.max)]
div_tog_oom = bind_cols(load_oom = g_s_load_oom, shannon_oom = shannon_div_oom , simpson_oom = simpson_div_oom, J_oom = J_oom, max_tax_oom = max_tax_oom)
```

###Combine div_tog
```{r}
div_tog_all = bind_rows(div_tog_bac, div_tog_oom)
```

###Now let's build models to test the relationship
```{r}

load_max_bac = betareg(max_tax_bac ~ load_bac, data = div_tog_bac)
load_max_oom = betareg(max_tax_oom ~ log10(load_oom), data = div_tog_oom)

summary(load_max_bac)
summary(load_max_oom)
```


###Now we can build the plots
```{r}
J_plot_bac<-ggplot(data=div_tog, aes(x=log10(load_bac), y=J)) + 
  geom_point(pch=20) + 
  geom_smooth(method='lm', color='red', level=0) +
  theme_bw() + 
  theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  xlab(expression(paste(' Load (log'[10],"(Total Microb. Cov./Plant Cov.))", sep=""))) +
  ylab("Evenness (Pielou's J)")

J_plot_oom<-ggplot(data=div_tog, aes(x=log10(load_oom), y=J)) + 
  geom_point(pch=20) + 
  geom_smooth(method='lm', color='red', level=0) +
  theme_bw() + 
  theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  xlab(expression(paste(' Load (log'[10],"(Total Microb. Cov./Plant Cov.))", sep=""))) +
  ylab("Evenness (Pielou's J)")

Shannon_div_plot_bac<-ggplot(data=div_tog, aes(x=load_bac, y=shannon)) + 
  geom_point(pch=20) + 
  geom_smooth(method='lm', color='red', level=0) +
  theme_bw() + 
  theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  xlab("Total Load") +
  #xlab(expression(paste(' Load (log'[10],"(Total Microb. Cov./Plant Cov.))", sep=""))) +
  ylab("Shannon Index (H')")

Shannon_div_plot_oom<-ggplot(data=div_tog, aes(x=log10(load_oom), y=shannon)) + 
  geom_point(pch=20) + 
  geom_smooth(method='lm', color='red', level=0) +
  theme_bw() + 
  theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  xlab(expression(paste(' Load (log'[10],"(Total Microb. Cov./Plant Cov.))", sep=""))) +
  ylab("Shannon Index (H')")

max_plot_bac<-ggplot(data=div_tog, aes(x=(load_bac), y=max_tax_bac)) + 
  geom_point(pch=20) + 
  #geom_smooth(method='lm', color='red', level=0) +
  geom_line(aes(y = predict(load_max_bac, div_tog_bac)), colour = "blue") +
  theme_bw() + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  xlab("Total Load") +
  #xlab(expression(paste(' Load (log'[10],"(Total Microb. Cov./Plant Cov.))", sep=""))) +
  ylab("Maximum Abundance (% of total)") + 
  scale_y_continuous(labels = function(x) x * 100) +
  geom_text(aes(label=ifelse(load_bac>4,as.character(max_tax_names),'')),position=position_jitter(width=.1,height=.1))

max_plot_oom<-ggplot(data=div_tog_oom, aes(x=log10(load_oom), y=max_tax_oom)) + 
  geom_point(pch=20) + 
  #geom_smooth(method='lm', color='red', level=0) +
  geom_line(aes(y = predict(load_max_oom, div_tog_oom)), colour = "blue") +
  theme_bw() + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  xlab(expression(paste(' Load (log'[10],"(Total Microb. Cov./Plant Cov.))", sep=""))) +
  ylab("Maximum Abundance (% of total)") + 
  scale_y_continuous(labels = function(x) x * 100) +
  geom_text(aes(label=ifelse(load_oom>5,as.character(max_tax_names),'')),position=position_jitter(width=.1,height=.1))



plot_grid(Shannon_div_plot_bac, max_plot, ncol=1)
```

And plot to file
```{r}
pdf(paste(paste("~/Dropbox/controlled_metagenomics/results_figures/div_load", concat_name, sep=""),".pdf",sep=""), family = "ArialMT", useDingbats = FALSE)

plot_grid(Shannon_div_plot_bac, max_plot_bac, ncol=1)

dev.off()
```

###Does load affect the overall evenness of the sample?
```{r, echo = FALSE}
cor.test(g_s_load_bacteria, J)

cor.test(g_s_load_bacteria, shannon_div)

cor.test(g_s_load_bacteria, max_tax_bac)
```











```{r}
J_plot<-ggplot(data=div_tog, aes(x=log10(load), y=J)) + 
  geom_point(pch=20) + 
  geom_smooth(method = "glm", 
    method.args = list(family = quasibinomial(link = 'logit')), color='red', level=0) +
  theme_bw() + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  xlab("Load (Total Microb. Cov./Plant Cov.)") +
  ylab("Evenness (Pielou's J)")

Shannon_div_plot<-ggplot(data=div_tog, aes(x=log10(load), y=shannon)) + 
  geom_point(pch=20) + 
  geom_smooth(method='glm', method.args = list(family = quasibinomial(link = 'logit')), color='red', level=0) +
  theme_bw() + 
  theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  xlab(expression(paste(' Load (log'[10],"(Total Microb. Cov./Plant Cov.))", sep=""))) +
  ylab("Shannon Index (H')")

max_plot<-ggplot(data=div_tog, aes(x=log10(load), y=max_tax)) + 
  geom_point(pch=20) + 
  geom_smooth(method = "glm", 
    method.args = list(family = quasibinomial(link = 'logit')), color='red', level=0) +
  theme_bw() + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  xlab(expression(paste(' Load (log'[10],"(Total Microb. Cov./Plant Cov.))", sep=""))) +
  ylab("Maximum Abundance (% of total)")  + 
  scale_y_continuous(labels = function(x) x * 100) +
  geom_text(aes(label=ifelse(load>0.05,as.character(max_tax_names),'')),position=position_jitter(width=.3,height=.3))

plot_grid(J_plot, max_plot, ncol=1)

pdf(paste(paste("~/Dropbox/controlled_metagenomics/results_figures/div_load", concat_name, sep=""),".pdf",sep=""), family = "ArialMT")
plot_grid(J_plot, max_plot, ncol=1)
dev.off()
```

























##Fungi
```{r, echo=FALSE}
concat_name="_fungi.csv"

g_s_fungi=read.csv(paste("~/Dropbox/controlled_metagenomics/results_figures/sweden_germany_combined", concat_name, sep=''), header = TRUE, row.names = 1, stringsAsFactors = FALSE)

g_s_tot_fungi = process_gs(g_s_fungi, subset=FALSE)

g_s_otus_filter_fungi = g_s_tot_fungi[[1]]

g_s_country_fungi = g_s_tot_fungi[[2]]

g_s_pop_fungi = g_s_fungi[[3]]

g_s_load_fungi = g_s_tot_fungi[[4]]

max_tax_fungi=apply(g_s_otus_filter_fungi, 1, max)
max_tax_names_fungi = colnames(g_s_otus_filter_fungi)[as.integer(apply(g_s_otus_filter_fungi, 1, which.max))]

```


###Merge the max taxes
```{r}
full_join(data.frame(max_tax_bac), data.frame(max_tax_fungi))
```


# Average genome coverage:
```{r, echo = FALSE}
g_s_otus_bac <- g_s_bacteria[,which(colnames(g_s_bacteria)%ni%c("population", "country", "load"))]
g_s_otus_oom <- g_s_oomycete[,which(colnames(g_s_oomycete)%ni%c("population", "country", "load"))]
g_s_otus_fungi <- g_s_fungi[,which(colnames(g_s_fungi)%ni%c("population", "country", "load"))]
bac_load = data.frame(rowSums(g_s_otus_bac))/2
oom_load = data.frame(rowSums(g_s_otus_oom))
fungi_load = data.frame(rowSums(g_s_otus_fungi))
names(bac_load) = "Bac_genome_cov"
names(oom_load) = "Oom_genome_cov"
names(fungi_load) = "Fun_genome_cov"

load_hist_bac <-ggplot(data = bac_load, aes(x = Bac_genome_cov)) +
  geom_histogram( bins = 40, fill = "#1B9E77") + 
  theme_bw() +
  theme(axis.text.x = element_text(angle = 0, hjust = 1), panel.border = element_rect(colour = "black", fill=NA, size= .5)) +
  scale_x_continuous(breaks = c(0:7)) +
  xlab("Estimated Bacterial Cells/Plant Cell") +
  annotate("text", x = 5.5, y = 55, label = c("Bacteria"), cex = 8)

load_hist_oom <-ggplot(data = oom_load, aes(x = Oom_genome_cov)) +
  geom_histogram( bins = 40, fill = "#1B9E77") + 
  theme_bw() +
  theme(axis.text.x = element_text(angle = 0, hjust = 1), panel.border = element_rect(colour = "black", fill=NA, size= .5)) +
  scale_x_continuous(breaks = c(0:10)/10) +
  xlab("Estimated Oomycete Chromosome Coverage/Plant Coverage") +
  annotate("text", x = .55, y = 175, label = c("Oomycete"), cex = 8)

load_hist_fungi <-ggplot(data = fungi_load, aes(x = Fun_genome_cov)) +
  geom_histogram( bins = 40, fill = "#1B9E77") + 
  theme_bw() +
  theme(axis.text.x = element_text(angle = 0, hjust = 1), panel.border = element_rect(colour = "black", fill = NA, size= .5)) +
  scale_x_continuous(breaks = c(0:10)/100) +
  xlab("Estimated Fungal Chromosome Coverage/Plant Coverage") +
  annotate("text", x = .03, y = 30, label = c("Fungus"), cex = 8)

pdf("~/Dropbox/controlled_metagenomics/results_figures/mean_load_oom_bac_fungus.pdf", family = "ArialMT", height = 10, width = 5)
plot_grid(load_hist_bac, load_hist_oom, load_hist_fungi, nrow = 3)
dev.off()
```

###Basic statistics on the mean and median abundances of microbes.
```{r, echo = FALSE}
median(bac_load$Bac_genome_cov)

mean(bac_load$Bac_genome_cov)

var(bac_load$Bac_genome_cov)
```

###Does load differ between locations?
```{r}
g_s_all_info = bind_cols(country = g_s_country, pop = g_s_pop_bacteria, load_bac = g_s_load_bacteria, load_fungi = g_s_load_fungi, load_oom = g_s_load_oomycete)

g_s_all_info = g_s_all_info[which(is.na(g_s_all_info$pop)==FALSE),]

wilcox.test(g_s_all_info[which(g_s_all_info$country == "germany"),]$load_bac, g_s_all_info[which(g_s_all_info$country == "Sweden"),]$load_bac)

wilcox.test(g_s_all_info[which(g_s_all_info$country == "germany"),]$load_fungi, g_s_all_info[which(g_s_all_info$country == "Sweden"),]$load_fungi)

wilcox.test(g_s_all_info[which(g_s_all_info$country == "germany"),]$load_oom, g_s_all_info[which(g_s_all_info$country == "Sweden"),]$load_oom)
```


What is the effect of population on load?
#```{r}
m2 <- lmer(data=g_s_bac_all_info, load ~ 1|country/pop)
m1 <- update(m2, .~ 1|country)

anova(m1, m2)
#```



```