---
title: "Microbiome analysis of Germany and Sweden"
author: "Talia_Karasov"
date: "5/3/2019"
output: html_document
---
```{r global_options, include=FALSE}
knitr::opts_chunk$set(fig.width=12, fig.height=8, fig.path='Figs/',
                      echo=FALSE, warning=FALSE, message=FALSE)
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(reshape2)
library('dplyr')
library('tidyr')
library(ggplot2)
library(RColorBrewer)
library(wesanderson)
library(matrixStats)
require(gridExtra)
library(cowplot)
#library(moments)
library(intrval)
library("Hmisc")
library(ape)
library(cowplot)
library(vegan)
library(extrafont)
library(lme4)
library(ape)
#font_import()
loadfonts()
manual_colors1=c("darkblue", "darkgoldenrod1", "darkseagreen", "darkorchid", "darkolivegreen1", "lightskyblue", "darkgreen", "firebrick", "brown1", "darkorange1") 
manual_colors2 = c("cyan1", "royalblue4", "darksalmon", "darkblue", "royalblue4", "dodgerblue3", "steelblue1", "lightskyblue", "darkseagreen", "darkgoldenrod1", "darkseagreen", "darkorchid", "darkolivegreen1", "brown1", "darkorange1", "cyan1", "darkgrey", "deeppink", "khaki2")
source("~/Dropbox/controlled_metagenomics/scripts/keep_used_in_publication/controlled_metagenomics_functions.R")
```

#Load vs. microbiome composition
This script explores the relationship between load and microbiome for the german and swedish datasets and graphs the microbiomes of these datasets


##Overall Microbiome Comparison
The goal of this next part is to generate figures side-by-side that will take the metagenome corrected tables for germany and sweden and graph the bar plots next to one another

First thing is to remove the offending columns from the german dataset
```{r, echo = FALSE}
german_samples=read.table("~/Dropbox/germany_pathogen_collections/sample_data/plate_sample_locations/sample_infoFinal_2018.txt", sep="\t", header=T)
keep=german_samples[-which(duplicated(german_samples$uniqueID)),]
```

Now we want to generate the side by side histograms of loads
```{r, echo = FALSE}
out_bacteria=meta_table("bacteria")
p_bac=plot_side_by_side(out_bacteria, colors = sample(RColorBrewer::brewer.pal(n=9, 'Paired'),9))

out_fungi=meta_table("fungi")
p_fungi=plot_side_by_side(out_fungi, colors = sample(RColorBrewer::brewer.pal(n=9, 'Paired'),9))

out_oom=meta_table("oomycete")
p_oom=plot_side_by_side(out_oom, colors = sample(RColorBrewer::brewer.pal(n=8, 'Dark2'),8))
```

```{r }
p_tot=plot_grid(p_bac, p_oom, ncol=2, align = 'h')
p_tot

pdf("~/Dropbox/controlled_metagenomics/results_figures/oomycete_bacteria_germany_sweden.pdf", family = "ArialMT", width = 8, height = 5)
p_tot
dev.off()

pdf("~/Dropbox/controlled_metagenomics/results_figures/oomycete_germany_sweden.pdf", family = "ArialMT", height = 5, width = 5)
p_oom
dev.off()

pdf("~/Dropbox/controlled_metagenomics/results_figures/bacteria_germany_sweden.pdf", family = "ArialMT", height = 5, width = 5)
p_bac
dev.off()

pdf("~/Dropbox/controlled_metagenomics/results_figures/fungi_germany_sweden.pdf", family = "ArialMT", height = 5, width = 5)
p_fungi
dev.off()

```

##Bacteria
First we will look at those reads classified as bacterial. We will perform principal coordinate analysis on the bray-curtis disimilarity to assess how load relates to similarity of microbiome content.

```{r, echo=FALSE}

concat_name="_bacteria.csv"

g_s=read.csv(paste("~/Dropbox/controlled_metagenomics/results_figures/sweden_germany_combined", concat_name, sep=''), header = TRUE, row.names = 1, stringsAsFactors = FALSE)

g_s_tot = process_gs(g_s, subset=FALSE)

g_s_otus_filter = g_s_tot[[1]]

g_s_country = g_s_tot[[2]]

g_s_pop = g_s_tot[[3]]

g_s_load = g_s_tot[[4]]

bray_curtis <-as.matrix(vegdist(g_s_otus_filter, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE))
```

###Which microbe has the highest mean load across samples in a region?
```{r}
find_top_abundance(g_s_otus_filter)
```

###PCoA on bray-curtis
```{r}
bc_pcoa=pcoa(bray_curtis, correction = "lingoes")
perc_explained=bc_pcoa$values[,3]
```

####Load vs PCs continued
```{r}
load_pcoa <- ggplot(data=data.frame(bc_pcoa$vectors), aes(x=Axis.1, y=Axis.2)) + 
  geom_point(aes(color=log10(g_s_load)), cex = 2) +
  scale_color_gradient2(name = "", low = "blue", mid = "white", high = "red") +
  theme_bw() + theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank(), text = element_text(size = 20)) +
  xlab("") + 
  theme(legend.key.width=unit(1,"cm"), axis.title.y = element_blank(), legend.text.align = 1)
# ylab(paste(paste("Axis 2:", (100*round(perc_explained[2],3)), sep=" "), "%", sep=""))

pop_pcoa <-ggplot(data=data.frame(bc_pcoa$vectors), aes(x=Axis.1, y=Axis.2)) + 
  geom_point(aes(color=g_s_country), cex = 2) +
  scale_color_discrete(name="Country") +
  theme_bw() + theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank(), text = element_text(size = 20)) +
  xlab(paste(paste("Axis 1:", (100*round(perc_explained[1],3)), sep=" "), "%", sep="")) +
 theme(legend.key.width=unit(1,"cm"), axis.title.y = element_blank()) 
#  ylab(paste(paste("Axis 2:", (100*round(perc_explained[2],3)), sep=" "), "%", sep=""))

gA <- ggplotGrob(load_pcoa)
gB <- ggplotGrob(pop_pcoa)
gA$widths <- gB$widths

grid.arrange(gA, gB, nrow = 2, left = paste(paste("Axis 2:", (100*round(perc_explained[2],3)), sep=" "), "%", sep=""))

pdf(paste(paste("~/Dropbox/controlled_metagenomics/results_figures/pcoa_bray_curtis_", concat_name, sep=""), ".pdf", sep=""), family = "ArialMT", useDingbats=FALSE)
grid.arrange(gA, gB, nrow = 2, left = paste(paste("Axis 2:", (100*round(perc_explained[2],3)), sep=" "), "%", sep=""), widths = 10)
dev.off()
```


###Do PC1 and PC2 distinguish between samples from germany vs sweden?
First PC1:
```{r, echo = FALSE}
pc1<-bc_pcoa$vectors[,1]
german_pc1<-pc1[g_s_country=="germany"]
sweden_pc1<-pc1[g_s_country=="Sweden"]
wilcox.test(german_pc1, sweden_pc1)
#p=0.1486 W=5367
```

###Next PC2:
```{r, echo = FALSE}
pc2<-bc_pcoa$vectors[,2]
german_pc2<-pc2[g_s_country=="germany"]
sweden_pc2<-pc2[g_s_country=="Sweden"]
wilcox.test(german_pc2, sweden_pc2)
#p=2.2e-16 W=8645
```

###What is the relationship between load and PC1?
```{r}
cor.test(g_s_load, pc1)
```

###How is load associated with Shannon Diversity

shannon_div <- vegan::diversity(g_s_otus_filter, index = "shannon")
simpson_div <- vegan::diversity(g_s_otus_filter, index="simpson")
J <- shannon_div/log(specnumber(g_s_otus_filter))
max_tax=apply(g_s_otus_filter, 1, max)
max_tax_names = colnames(g_s_otus_filter)[apply(g_s_otus_filter, 1, which.max)]

div_tog=bind_cols(load=g_s_load, shannon=shannon_div, simpson=simpson_div, J=J, max_tax=max_tax)

J_plot<-ggplot(data=div_tog, aes(x=log10(load), y=J)) + 
  geom_point(pch=20) + 
  geom_smooth(method = "glm", method.args = list(family = quasibinomial(link = 'logit')), color='red', level=0) +
  theme_bw() + 
  theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  xlab(expression(paste('Load (log'[10],"(Total Microb. Cov./Plant Cov.))", sep=""))) +
  ylab("Evenness (Pielou's J)")

Shannon_div_plot<-ggplot(data=div_tog, aes(x=lload, y=shannon)) + 
  geom_point(pch=20) + 
  geom_smooth(method = "glm", method.args = list(family = binomial(link = 'logit')), color='red', level=0) +
  theme_bw() + 
  theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  xlab(expression(paste(' Load (log'[10],"(Total Microb. Cov./Plant Cov.))", sep=""))) +
  ylab("Shannon Index (H')")

max_plot<-ggplot(data=div_tog, aes(x=log10(load), y=max_tax)) + 
  geom_point(pch=20) + 
  geom_smooth(method = "glm", method.args = list(family = quasibinomial(link = 'logit')), color='red', level=0) +
  theme_bw() + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  xlab(expression(paste(' Load (log'[10],"(Total Microb. Cov./Plant Cov.))", sep=""))) +
  ylab("Maximum Abundance (% of total)") + 
  scale_y_continuous(labels = function(x) x * 100) +
  geom_text(aes(label=ifelse(load>5,as.character(max_tax_names),'')),position=position_jitter(width=.1,height=.1))


plot_grid(J_plot, max_plot, ncol=1)

pdf(paste(paste("~/Dropbox/controlled_metagenomics/results_figures/div_load", concat_name, sep=""),".pdf",sep=""), family = "ArialMT")

plot_grid(J_plot, max_plot, ncol=1)

dev.off()
```

###Does load affect the overall evenness of the sample?
#```{r}
cor.test(g_s_load, J)

cor.test(g_s_load, shannon_div)
```

##Oomycetes

```{r, echo=FALSE}
concat_name="_oomycete.csv"

g_s=read.csv(paste("~/Dropbox/controlled_metagenomics/results_figures/sweden_germany_combined", concat_name, sep=''), header = TRUE, row.names = 1, stringsAsFactors = FALSE)

g_s_tot = process_gs(g_s, subset=FALSE)

g_s_otus_filter = g_s_tot[[1]]

g_s_country = g_s_tot[[2]]

g_s_pop = g_s_tot[[3]]

g_s_load = g_s_tot[[4]]

bray_curtis <-as.matrix(vegdist(g_s_otus_filter, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE))

#PCoA on bray-curtis
bc_pcoa=pcoa(bray_curtis, correction = "lingoes")

perc_explained=bc_pcoa$values[,3]

####Load vs PCs continued
load_pcoa <- ggplot(data=data.frame(bc_pcoa$vectors), aes(x=Axis.1, y=Axis.2)) + 
  geom_point(aes(color=log10(g_s_load)), cex = 2) +
  scale_color_gradient2(name = "", low = "blue", mid = "white", high = "red") +
  theme_bw() + theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank(), text = element_text(size = 20)) +
  xlab("") + 
  theme(legend.key.width=unit(1,"cm"), axis.title.y = element_blank(), legend.text.align = 1)
# ylab(paste(paste("Axis 2:", (100*round(perc_explained[2],3)), sep=" "), "%", sep=""))

pop_pcoa <-ggplot(data=data.frame(bc_pcoa$vectors), aes(x=Axis.1, y=Axis.2)) + 
  geom_point(aes(color=g_s_country), cex = 2) +
  scale_color_discrete(name="Country") +
  theme_bw() + theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank(), text = element_text(size = 20)) +
  xlab(paste(paste("Axis 1:", (100*round(perc_explained[1],3)), sep=" "), "%", sep="")) +
 theme(legend.key.width=unit(1,"cm"), axis.title.y = element_blank()) 
#  ylab(paste(paste("Axis 2:", (100*round(perc_explained[2],3)), sep=" "), "%", sep=""))

gA <- ggplotGrob(load_pcoa)
gB <- ggplotGrob(pop_pcoa)
gA$widths <- gB$widths
gA <- ggplotGrob(load_pcoa)
gB <- ggplotGrob(pop_pcoa)
gA$widths <- gB$widths

grid.arrange(gA, gB, nrow = 2, left = paste(paste("Axis 2:", (100*round(perc_explained[2],3)), sep=" "), "%", sep=""))

pdf(paste(paste("~/Dropbox/controlled_metagenomics/results_figures/pcoa_bray_curtis_", concat_name, sep=""), ".pdf", sep=""), family = "ArialMT", useDingbats = FALSE)
grid.arrange(gA, gB, nrow = 2, left = paste(paste("Axis 2:", (100*round(perc_explained[2],3)), sep=" "), "%", sep=""))
dev.off()

```

###Which microbe has the highest load across samples?
```{r}
find_top_abundance(g_s_otus_filter)

```

###Do PC1 and PC2 distinguish between samples from germany vs sweden?
First PC1:
```{r, echo = FALSE}
pc1<-bc_pcoa$vectors[,1]
german_pc1<-pc1[g_s_country=="germany"]
sweden_pc1<-pc1[g_s_country=="Sweden"]
wilcox.test(german_pc1, sweden_pc1)
#p=0.0033 W=5367
```

###Next PC2:
```{r, echo = FALSE}
pc2<-bc_pcoa$vectors[,2]
german_pc2<-pc2[g_s_country=="germany"]
sweden_pc2<-pc2[g_s_country=="Sweden"]
wilcox.test(german_pc2, sweden_pc2)
#p=0.0309 W=5694
```

###What is the relationship between load and PC1?
```{r, echo = FALSE}
cor.test(g_s_load, pc1)
```

###How is load associated with Shannon Diversity
#```{r, echo=FALSE}
shannon_div <- vegan::diversity(g_s_otus_filter, index = "shannon")
simpson_div <- vegan::diversity(g_s_otus_filter, index="simpson")
J <- shannon_div/log(specnumber(g_s_otus_filter))
max_tax=apply(g_s_otus_filter, 1, max)
max_tax_names = colnames(g_s_otus_filter)[apply(g_s_otus_filter, 1, which.max)]
div_tog=bind_cols(load=g_s_load, shannon=shannon_div, simpson=simpson_div, J=J, max_tax=max_tax)

J_plot<-ggplot(data=div_tog, aes(x=log10(load), y=J)) + 
  geom_point(pch=20) + 
  geom_smooth(method = "glm", 
    method.args = list(family = quasibinomial(link = 'logit')), color='red', level=0) +
  theme_bw() + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  xlab("Load (Total Microb. Cov./Plant Cov.)") +
  ylab("Evenness (Pielou's J)")

Shannon_div_plot<-ggplot(data=div_tog, aes(x=log10(load), y=shannon)) + 
  geom_point(pch=20) + 
  geom_smooth(method='glm', method.args = list(family = quasibinomial(link = 'logit')), color='red', level=0) +
  theme_bw() + 
  theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  xlab(expression(paste(' Load (log'[10],"(Total Microb. Cov./Plant Cov.))", sep=""))) +
  ylab("Shannon Index (H')")

max_plot<-ggplot(data=div_tog, aes(x=log10(load), y=max_tax)) + 
  geom_point(pch=20) + 
  geom_smooth(method = "glm", 
    method.args = list(family = quasibinomial(link = 'logit')), color='red', level=0) +
  theme_bw() + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  xlab(expression(paste(' Load (log'[10],"(Total Microb. Cov./Plant Cov.))", sep=""))) +
  ylab("Maximum Abundance (% of total)")  + 
  scale_y_continuous(labels = function(x) x * 100) +
  geom_text(aes(label=ifelse(load>0.05,as.character(max_tax_names),'')),position=position_jitter(width=.3,height=.3))

plot_grid(J_plot, max_plot, ncol=1)

pdf(paste(paste("~/Dropbox/controlled_metagenomics/results_figures/div_load", concat_name, sep=""),".pdf",sep=""), family = "ArialMT")
plot_grid(J_plot, max_plot, ncol=1)
dev.off()
```

###Does load affect the overall evenness of the sample?
#```{r, echo = FALSE}
cor.test(g_s_load, J)

cor.test(g_s_load, shannon_div)
```
