---
title: "DC3000 controlled_infections"
author: "Talia_Karasov"
date: "5/6/2019"
output: html_document
---
```{r global_options, include=FALSE}
knitr::opts_chunk$set(fig.width=12, fig.height=8, fig.path='Figs/',
                      echo=FALSE, warning=FALSE, message=FALSE)
```

```{r setup, include=FALSE}
library(reshape2)
library('dplyr')
library('tidyr')
library(ggplot2)
library(RColorBrewer)
library(wesanderson)
require(gridExtra)
library(cowplot)
library(lmtest)
library(Hmisc)
library(igraph)
library(extrafont)
'%!in%' <- function(x,y)!('%in%'(x,y))
font_import()
mycolors=c("darkblue", "darkgoldenrod1", "darkseagreen", "darkorchid", "darkolivegreen1", "lightskyblue", "firebrick", "khaki2", "brown1", "darkorange1", "cyan1", "royalblue4", "darksalmon", "darkblue","royalblue4", "dodgerblue3", "steelblue1", "lightskyblue", "darkseagreen", "darkgoldenrod1", "darkseagreen", "darkorchid", "darkolivegreen1", "brown1", "darkorange1", "cyan1", "darkgrey")

replace_rr <- function(rr){
  if(is.na(rr)) return(NA)
  else if(rr=="I") return(0)
  else if(rr=="II") return(1)
  else if(rr=="III") return(2)
  else if(rr=="IV") return(3)
  else if(rr=="V") return(4)
  else if(rr=="VI") return(6)
  else return(NA)
}

g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)
}


#meta=read.table("~/work_main/abt6_projects9/metagenomic_controlled/data/processed_reads/dc3000_infections/meta_family_corrected_per_plant.csv", sep=",", header=T, row.names = 1)
meta=read.table("~/work_main/abt6_projects9/metagenomic_controlled/data/processed_reads/dc3000_infections/meta_family_corrected_per_plant.csv", sep=",", header=T, row.names = 1)

top10=names(sort(rowSums(meta, na.rm=TRUE), decreasing=TRUE )[1:10])
meta_microbiome=meta[top10,]
rest=colSums(meta[rownames(meta) %!in% top10, ], na.rm=TRUE)
meta_microbiome=rbind(meta_microbiome, rest)
rownames(meta_microbiome)[11]="Rest"
tot=colSums(meta_microbiome)
meta_microbiome$Family=rownames(meta_microbiome)
microb_melt=melt(meta_microbiome, id=c("Family"))
microb_melt$day_ori=sapply(strsplit(as.character(microb_melt$variable), "_"), `[`, 2)
microb_melt$Genotype=sapply(strsplit(as.character(microb_melt$variable), "_"), `[`,1)
microb_melt$Day=sapply(strsplit(microb_melt$day_ori, "\\."), `[`, 1)
microb_melt$Day=gsub("R","", microb_melt$Day)
microb_melt$Day=sapply(microb_melt$Day, replace_rr)
#microb_melt$Genotype=gsub("avrB", "Nope", microb_melt$Genotype)
microb_melt$Replicate=sapply(strsplit(microb_melt$day_ori, "\\."), `[`, 2)
microb_melt$Replicate=gsub( "_","", microb_melt$Replicate)
microb_melt$combined=paste(microb_melt$Day, microb_melt$Replicate, sep="_")
```


####Plot the relationship between Colony Counting, qPCR and metagenomics
```{r, echo=FALSE}
# Colony Count Data
colony_count=read.table("~/Dropbox/controlled_metagenomics/infection_results/dc3000_colony_count.txt", sep="\t")
colony_count$Genotype=gsub(" ", "",colony_count$Genotype)

#qPCR data
qPCR = read.table("~/Dropbox/controlled_metagenomics/qPCR/qPCR_delta.txt", sep="\t")
colnames(qPCR)[1:3]=c("Genotype", "Day", "Replicate")
qPCR=qPCR[, c("Day", "Replicate", "Genotype",  "ratio")]

#Combine colony count and qPCR
combined = dplyr::full_join(colony_count, qPCR)

#ggplot(data=combined, aes(x=log10(FINAL+.00001), y=log10(ratio+0.00001))) +  geom_point() + xlab(expression(log[10]~(cfu/mm))) + ylab(expression(log[10]~("ng bacteria"/"ng plant"))) + geom_smooth(method=lm, se = FALSE, col="Gray")
       
#cor.test(log10(combined$ratio+0.000001), log10(combined$FINAL+.000001))

#metagenomics data
meta=read.table("~/work_main/abt6_projects9/metagenomic_controlled/data/processed_reads/dc3000_infections/meta_family_corrected_per_plant.csv", sep=",", header=T, row.names = 1)
meta_pseud=melt(meta[c("Pseudomonadaceae"),], value.name="load", variable.name="Genotype")
#meta_pseud$Genotype=rownames(meta_pseud)
meta_pseud$day_ori=sapply(strsplit(as.character(meta_pseud$Genotype), "-"), `[`, 1)
meta_pseud$Genotype=sapply(strsplit(as.character(meta_pseud$Genotype), "_"), `[`,1)
meta_pseud$Day=sapply(strsplit(sapply(strsplit(meta_pseud$day_ori, "\\."), `[`, 1), "_R"), `[`,2)
meta_pseud$Replicate=sapply(strsplit(meta_pseud$day_ori, "\\."), `[`, 2)
meta_pseud$Replicate=gsub( "_","", meta_pseud$Replicate)
meta_pseud$Day=sapply(meta_pseud$Day, replace_rr)

#combine all three methods
all_all=merge(combined,meta_pseud)
all_keep=all_all[which(all_all$ratio!="NA"),]

#ggplot(data=all_keep, aes(x=log10(FINAL+1), y=log10(load))) +  geom_point() + xlab(expression(log[10]~(cfu/mm))) + ylab(expression(log[10]~("ng bacteria"/"ng plant"))) + geom_smooth(method=lm, se = FALSE, col="Gray")

#now build plot for all three
load_plot=all_keep%>% group_by(Day, Genotype) %>% summarise(mean = mean(log10(load+0.0001), na.rm=T), se = sd(log10(load+.0001), na.rm=T) / length(load)) 

cfu_plot=all_keep%>% group_by(Day, Genotype) %>% summarise(mean = mean(log10(FINAL+1), na.rm=T), se = sd(log10(FINAL+1), na.rm=T) / length(FINAL)) 

qpcr_plot=all_keep%>% group_by(Day, Genotype) %>% summarise(mean = mean(log10(ratio+0.0001), na.rm=T), se = sd(log10(ratio+0.0001), na.rm=T) / length(ratio)) 

#Bind the three together
df <- bind_rows(CFU=cfu_plot, qPCR=qpcr_plot, Metagenome=load_plot, .id="group")

pdf("~/Dropbox/controlled_metagenomics/results_figures/tri_comparison.pdf", useDingbats = FALSE, family = "ArialMT")
ggplot(df, aes(x=Day, y=mean, group=Genotype, colour=Genotype))+geom_errorbar(aes(ymin=mean-se, ymax=mean+se), width=.1) +
  geom_point()+theme_bw()+geom_line()+theme_bw()+ 
  facet_wrap(~group , scales = "free_y", ncol=1, strip.position = "right" )             + 
  ylab(NULL) + theme(legend.position = "none") + scale_color_manual(values = wes_palette(n=3, name="GrandBudapest1"))
dev.off()
```


####Calculate correlations between methods


```{r, echo=FALSE}
pdf("~/Dropbox/controlled_metagenomics/results_figures/cfu_vs_other.pdf", height=12, width=6, useDingbats = FALSE, family = "ArialMT")
plot1 <- ggplot(data=all_keep, aes(x=log10(FINAL+1), y=log10(ratio+0.01)))+geom_point()  + xlab(expression(log[10]~(cfu/mm))) + ylab(expression(log[10]~("ng bact."/"ng plant"))) + geom_smooth(method=lm, se = FALSE, col="Gray") + theme_bw()
plot2 <- ggplot(data=all_keep, aes(x=log10(FINAL+1), y=log10(load+0.01)))+geom_point() + xlab(expression(log[10]~(cfu/mm))) + ylab(expression(log[10]~("DC3000 coverage"/"Plant coverage"))) + geom_smooth(method=lm, se = FALSE, col="Gray") +theme_bw()
plot3 <- ggplot(data=all_keep, aes(x=log10(ratio+0.01), y=log10(load+0.01)))+geom_point() + ylab(expression(log[10]~("DC3000 coverage"/"Plant coverage"))) + xlab(expression(log[10]~("ng bact."/"ng plant"))) + geom_smooth(method=lm, se = FALSE, col="Gray") +theme_bw()
grid.arrange(plot1, plot2,plot3, ncol=1)
dev.off()
```

####What is the correlation between qPCR and colony counting?
```{r, echo = FALSE}
cor.test(log10(all_keep$ratio+0.01), log10(all_keep$FINAL+1))
```

####What is the correlation between load and colony counting?
```{r, echo = FALSE}
cor.test(log10(all_keep$load+0.01), log10(all_keep$FINAL+1))
```

####What is the mean and variance of load for DC3000 each day?
```{r, echo = FALSE}
per_day = all_keep %>% group_by(Day, Genotype) 
mean_per_day = summarise(per_day, load_mean = mean(log10(load)), sd_load = sd(log10(load)))
```


####Now for plotting changes in the microbiome over time
```{r, echo=FALSE}
microb_melt$Gen_Day = paste(microb_melt$Genotype, microb_melt$Day, sep="_")
# microb_melt$Day2=relevel(microb_melt$Day, "0")
# microb_melt$Genotype2=relevel(as.factor(microb_melt$Genotype), "C")

#ONLY EV
EV=microb_melt[microb_melt$Genotype=="EV",]
EV=EV[which(is.na(EV$Replicate)==FALSE),]
EV_plot <- ggplot(data=EV, aes(x=combined, y=log10(value+1), fill=Family))

#ONLY AVR
avrB=microb_melt[microb_melt$Genotype=="avrB",]
avrB=avrB[which(is.na(avrB$Replicate)==FALSE),]
avrB_plot <- ggplot(data=avrB, aes(x=combined, y=log10(value+1), fill=Family))

#ONLY CONTROL
control=microb_melt[microb_melt$Genotype=="C",]
control=control[which(is.na(control$Replicate)==FALSE),]
control_plot <- ggplot(data=control, aes(x=combined, y=log10(value+1), fill=Family))

#Final
avrB_final= avrB_plot + geom_bar(aes(), stat="identity", position="stack") +
  scale_fill_manual(values = mycolors) +
  theme(legend.position="none", panel.background = element_blank(), axis.line = element_line(colour = "black")) + guides(fill=guide_legend(nrow=5)) +
  xlab("") +  ylab(expression(log[10]~("Microbe cov."/"Plant cov.")))+ ylim(c(0,3))

control_final= control_plot + geom_bar(aes(), stat="identity", position="stack") +
  scale_fill_manual(values = mycolors) +
  theme(legend.position="none", panel.background = element_blank(), axis.line = element_line(colour = "black")) + guides(fill=guide_legend(nrow=5)) +
  xlab("") +  ylim(c(0,3)) + ylab("")

with_legend<- control_plot + geom_bar(aes(), stat="identity", position="stack") +
  scale_fill_manual(values = mycolors) +
  theme(legend.position="bottom", panel.background = element_blank(), axis.line = element_line(colour = "black")) + guides(fill=guide_legend(nrow=5)) +
  xlab("Plant Individuals") +  ylab(expression(log[10]~("Microbe cov."/"Plant cov."))) + ylim(c(0,3))

EV_final= EV_plot + geom_bar(aes(), stat="identity", position="stack") +
  scale_fill_manual(values = mycolors) +
  theme(legend.position="none", panel.background = element_blank(), axis.line = element_line(colour = "black")) + guides(fill=guide_legend(nrow=5)) +
  xlab("Day") + ylab("")+ ylim(c(0,3))

leg<-g_legend(with_legend)

#pdf("~/Dropbox/controlled_metagenomics/results_figures/dc3000_temporal_metagenome_Triplot.pdf", width=8, height=12, fonts = "Arial")
plot_grid(control_final, avrB_final, EV_final, leg, rel_heights=c(1/3.5,1/3.5,1/3.5, .5/3.5),  nrow=4)
#dev.off()

```

