---
title: "Microbiome analysis of Germany and Sweden"
author: "Talia_Karasov"
date: "5/3/2019"
output: html_document
---
```{r global_options, include=FALSE}
knitr::opts_chunk$set(fig.width=12, fig.height=8, fig.path='Figs/',
                      echo=FALSE, warning=FALSE, message=FALSE)
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(reshape2)
library('dplyr')
library('tidyr')
library(ggplot2)
library(RColorBrewer)
library(phyloseq)
library(matrixStats)
require(gridExtra)
library(cowplot)
#library(moments)
library(intrval)
library("Hmisc")
library(ape)
library(cowplot)
library(vegan)
library(extrafont)
library(lme4)
library(ape)
library(DESeq2)
#font_import()
loadfonts()
hue1_12 = c("#bf82cc","#5bc14d","#bf4db3","#9fba36","#7861ce","#4d8c28","#d83e76","#44c181","#d0452f","#4aadd6","#d6812c","#667fc7")
hue13_25 =c("#cbaa3b","#9c4769","#7dba6f","#dd809f","#3e8148","#c25d4d","#59c5b0","#de986d","#2f8a72","#91692e","#afb16c","#5f6c2b","#84892d")

source("~/Dropbox/controlled_metagenomics/scripts/keep_used_in_publication/controlled_metagenomics_functions.R")
```

```{r, echo=FALSE}

merge_phyloseq<-function(g_s){
  rownames(g_s) = g_s$Genotype
  g_s = g_s[,which(colnames(g_s)!="Genotype")]
  #g_s_otus <- g_s[,which(colnames(g_s)%ni%c("population", "country", "load", "Genotype"))]
  g_s_tot = process_gs(g_s, subset=FALSE)
  g_s_otus_comp = g_s_tot[[1]]
  g_s_country = g_s_tot[[2]]
  g_s_pop = g_s_tot[[3]]
  g_s_load = g_s_tot[[4]]
  
  #g_s_filter is the data that is not compositional but instead coverage per each genome
  g_s_filter = apply(round(g_s_tot[[5]] * 12000000), 2, as.integer)
  g_s_filter[is.na(g_s_filter)] = 0
  OTU = (otu_table(g_s_filter, taxa_are_rows = FALSE))
  row.names(OTU) = row.names(g_s_tot[[5]])
  
  #Prepare sample data
  sampledata = sample_data(data.frame(Population = g_s_pop, Country = g_s_country, load = g_s_load, row.names = row.names(OTU)))
  physeq = phyloseq(otu_table=OTU, sam_data=sampledata)
  return(physeq)
}


```

```{r}
# DESeq conversion 
# DESeq2 Variance Stabilization
# You must step through the size factor and dispersion estimates prior to calling the getVarianceStabilizedData() function.

deseq_variance_stab<-function(physeq, deseq_data){
  trial.DEseq = estimateSizeFactors(deseq_data)
  trial.DEseq = estimateDispersions(trial.DEseq)
  trial.vst = getVarianceStabilizedData(trial.DEseq)
  dim(trial.vst)

  # Save the untransformed data as a separate variable so you can go back to it
  trial.phyloseq0 = physeq

  # add the varience stabilized otu numbers into the dataset:
  otu_table(trial.phyloseq0) <- otu_table(trial.vst, taxa_are_rows = TRUE)
  
  return(trial.phyloseq0)
}
```


#Load vs. microbiome composition
This script explores the relationship between load and microbiome for the german and swedish datasets and graphs the microbiomes of these datasets


##Overall Microbiome Comparison
The goal of this next part is to generate figures side-by-side that will take the metagenome corrected tables for germany and sweden and graph the bar plots next to one another

First thing is to remove the offending columns from the german dataset
```{r, echo = FALSE}
german_samples=read.table("~/Dropbox/germany_pathogen_collections/sample_data/plate_sample_locations/sample_infoFinal_2018.txt", sep="\t", header=T)
keep=german_samples[-which(duplicated(german_samples$uniqueID)),]
```

##A. thaliana genome coverage
Instead of raw reads, we are using the metric of coverage per genome.
```{r}
athal_cov = read.csv("~/Dropbox/controlled_metagenomics/data/athaliana_cov_7_2019.csv", header = F, col.names = c("Genotype", "Coverage"))
athal = data.frame(athal_cov$Coverage)
rownames(athal) = athal_cov$Genotype

```

##Bacteria
First we will look at those reads classified as bacterial. We will perform principal coordinate analysis on the bray-curtis disimilarity to assess how load relates to similarity of microbiome content.

```{r}
concat_name="_bacteria.csv"
g_s1=read.csv(paste("~/Dropbox/controlled_metagenomics/data/sweden_germany_combined", concat_name, sep=''), header = TRUE, row.names = 1, stringsAsFactors = FALSE)
g_s1$Genotype = rownames(g_s1)
#Two samples are messed up
#g_s = g_s[which(rownames(g_s %ni% c("ULL.5", "Adal.5"), ]
#Merge the data table with the A. thaliana coverage
g_s = inner_join(athal_cov, g_s1)
physeq = merge_phyloseq(g_s)
phy_deseq_stab = deseq_variance_stab(physeq, deseq_country)

# Now, we re-do the ordination

```

But i think that most of PC1 is driven by low quality samples
```{r}
plot(x=(log10(otu_table(physeq)[,1])), y=pca_out$rotation[,1], pch = 20, xlab = "Genome Coverage for A. thaliana", ylab = "Loading on PC1")
#abline(v=0.5)
#cutoff at 0.26x coverage
```

But removing samples with low coverage doesn't fix the problem. With PCA, the results are just too influenced by the outliers. So let's give up on just doing PCA on the deseq samples. What about if i want to look at the differential loads between Sweden and Germany. I think we should do this with all of the reads readjusted.

```{r}
g_s1=read.csv(paste("~/Dropbox/controlled_metagenomics/data/sweden_germany_combined", concat_name, sep=''), header = TRUE, row.names = 1, stringsAsFactors = FALSE)
g_s1$Genotype = rownames(g_s1)
#Two samples are messed up
#g_s = g_s[which(rownames(g_s %ni% c("ULL.5", "Adal.5"), ]
#Merge the data table with the A. thaliana coverage
g_s = inner_join(athal_cov, g_s1)
g_s_otus <- g_s[,which(colnames(g_s)%ni%c("population", "country", "load", "Genotype"))]
physeq = merge_phyloseq(g_s)

```



```{r}
concat_name="_bacteria.csv"
g_s1=read.csv(paste("~/Dropbox/controlled_metagenomics/data/sweden_germany_combined", concat_name, sep=''), header = TRUE, row.names = 1, stringsAsFactors = FALSE)
g_s1$Genotype = rownames(g_s1)
#Two samples are messed up
#g_s = g_s[which(rownames(g_s %ni% c("ULL.5", "Adal.5"), ]
#Merge the data table with the A. thaliana coverage
g_s = inner_join(athal_cov, g_s1)
physeq = merge_phyloseq(g_s)

deseq_country = phyloseq_to_deseq2(physeq, ~Country)
country_wald = DESeq(deseq_country, test="Wald", fitType="mean")
res = results(country_wald, lfcThreshold = 1)
alpha = 0.01
sigtab = as(res[which(res$padj < alpha),], "data.frame")
sigtab$Family = rownames(sigtab)
```



```{r}
  deseq_country = phyloseq_to_deseq2(physeq, ~Country)
  country_wald = DESeq(deseq_country, test="Wald", fitType="parametric")
  res = results(country_wald, lfcThreshold= 1)
  alpha = 0.01
sigtab = as(res[which(res$padj < alpha),], "data.frame")
  sigtab$Family = rownames(sigtab)

#How many upregulated?
pos = sigtab[sigtab$log2FoldChange>1,] #10
neg = sigtab[sigtab$log2FoldChange<1,] #156
```  


Plot the differences
```{r}
theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
    scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab$log2FoldChange, sigtab$Family, function(x) max(x))
x = sort(x, TRUE)
sigtab$Family = factor(as.character(sigtab$Family), levels=names(x))
ggplot(sigtab, aes(x=Family, y=log2FoldChange, color=log10(pvalue))) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
```

```




````{r}
phy_deseq_stab = deseq_variance_stab(physeq, deseq_country)

# Now, we re-do the ordination
trial.ord <- ordinate(trial.phyloseq0, "CCA", autotransform=T)
plot_ordination(physeq, trial.ord, type = "samples", color = "Country")
plot_ordination(physeq, trial.ord, type = "samples", color = "load")

#What if we just do PCA on the 
pca_out = prcomp((otu_table(phy_deseq_stab)), scale = TRUE)
```

But i think that most of PC1 is driven by low quality samples
```{r}
plot(x=(log10(otu_table(physeq)[,1])), y=pca_out$rotation[,1], pch = 20, xlab = "Genome Coverage for A. thaliana", ylab = "Loading on PC1")
#abline(v=0.5)
#cutoff at 0.26x coverage
plot(pca_out$rotation[,1], pca_out$rotation[,2])
```


Bray Curits is not the best way to work with the transformed data due to the negative values. https://github.com/joey711/phyloseq/issues/492

###PCoA on bray-curtis
```{r}
perc_explained=pca_out$rotation
```

####Load vs PCs continued
```{r}
load_pca <- ggplot(data=data.frame(pca_out$rotation), aes(x=PC1, y=PC2)) + 
  geom_point(aes(color=log10(g_s_load)), cex = 2) +
  scale_color_gradient2(name = "", low = "blue", mid = "white", high = "red") +
  theme_bw() + theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank(), text = element_text(size = 20)) +
  xlab("") + 
  theme(legend.key.width=unit(1,"cm"), axis.title.y = element_blank(), legend.text.align = 1)
# ylab(paste(paste("Axis 2:", (100*round(perc_explained[2],3)), sep=" "), "%", sep=""))

pop_pcoa <-ggplot(data=data.frame(pca_out$rotation), aes(x=PC1, y=PC2)) + 
  geom_point(aes(color=g_s_country), cex = 2) +
  scale_color_discrete(name="Country") +
  theme_bw() + theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank(), text = element_text(size = 20)) +
  xlab(paste(paste("Axis 1:", (100*round(perc_explained[1],3)), sep=" "), "%", sep="")) +
 theme(legend.key.width=unit(1,"cm"), axis.title.y = element_blank()) 
#  ylab(paste(paste("Axis 2:", (100*round(perc_explained[2],3)), sep=" "), "%", sep=""))

gA <- ggplotGrob(load_pcoa)
gB <- ggplotGrob(pop_pcoa)
gA$widths <- gB$widths

grid.arrange(gA, gB, nrow = 2, left = paste(paste("Axis 2:", (100*round(perc_explained[2],3)), sep=" "), "%", sep=""))

pdf(paste(paste("~/Dropbox/controlled_metagenomics/results_figures/pcoa_bray_curtis_", concat_name, sep=""), ".pdf", sep=""), family = "ArialMT", useDingbats=FALSE)
grid.arrange(gA, gB, nrow = 2, left = paste(paste("Axis 2:", (100*round(perc_explained[2],3)), sep=" "), "%", sep=""), widths = 10)
dev.off()
```







#bray_curtis <-as.matrix(vegdist(deseq_country, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE))
```

###Which microbe has the highest mean load across samples in a region?
```{r}
find_top_abundance(g_s_otus_filter)
```

###PCoA on bray-curtis
```{r}
bc_pcoa=pcoa(bray_curtis, correction = "lingoes")
perc_explained=bc_pcoa$values[,3]
```

####Load vs PCs continued
```{r}
load_pcoa <- ggplot(data=data.frame(bc_pcoa$vectors), aes(x=Axis.1, y=Axis.2)) + 
  geom_point(aes(color=log10(g_s_load)), cex = 2) +
  scale_color_gradient2(name = "", low = "blue", mid = "white", high = "red") +
  theme_bw() + theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank(), text = element_text(size = 20)) +
  xlab("") + 
  theme(legend.key.width=unit(1,"cm"), axis.title.y = element_blank(), legend.text.align = 1)
# ylab(paste(paste("Axis 2:", (100*round(perc_explained[2],3)), sep=" "), "%", sep=""))

pop_pcoa <-ggplot(data=data.frame(bc_pcoa$vectors), aes(x=Axis.1, y=Axis.2)) + 
  geom_point(aes(color=g_s_country), cex = 2) +
  scale_color_discrete(name="Country") +
  theme_bw() + theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank(), text = element_text(size = 20)) +
  xlab(paste(paste("Axis 1:", (100*round(perc_explained[1],3)), sep=" "), "%", sep="")) +
 theme(legend.key.width=unit(1,"cm"), axis.title.y = element_blank()) 
#  ylab(paste(paste("Axis 2:", (100*round(perc_explained[2],3)), sep=" "), "%", sep=""))

gA <- ggplotGrob(load_pcoa)
gB <- ggplotGrob(pop_pcoa)
gA$widths <- gB$widths

grid.arrange(gA, gB, nrow = 2, left = paste(paste("Axis 2:", (100*round(perc_explained[2],3)), sep=" "), "%", sep=""))

pdf(paste(paste("~/Dropbox/controlled_metagenomics/results_figures/pcoa_bray_curtis_", concat_name, sep=""), ".pdf", sep=""), family = "ArialMT", useDingbats=FALSE)
grid.arrange(gA, gB, nrow = 2, left = paste(paste("Axis 2:", (100*round(perc_explained[2],3)), sep=" "), "%", sep=""), widths = 10)
dev.off()
```


###Do PC1 and PC2 distinguish between samples from germany vs sweden?
First PC1:
```{r, echo = FALSE}
pc1<-bc_pcoa$vectors[,1]
german_pc1<-pc1[g_s_country=="germany"]
sweden_pc1<-pc1[g_s_country=="Sweden"]
wilcox.test(german_pc1, sweden_pc1)
#p=0.1486 W=5367
```

###Next PC2:
```{r, echo = FALSE}
pc2<-bc_pcoa$vectors[,2]
german_pc2<-pc2[g_s_country=="germany"]
sweden_pc2<-pc2[g_s_country=="Sweden"]
wilcox.test(german_pc2, sweden_pc2)
#p=2.2e-16 W=8645
```

###What is the relationship between load and PC1?
```{r}
cor.test(g_s_load, pc1)
```


#PERMANOVA WITH BRAY-CURTIS DISTANCES
Here is some advice on nesting:
https://stats.stackexchange.com/questions/188519/adonis-in-vegan-order-of-variables-or-use-of-strata

```{r}
#exclude NA
bc_comp = bray_curtis[which(!is.na(g_s_pop)), which(!is.na(g_s_pop))]
g_s_load_com = g_s_load[which(!is.na(g_s_pop))]
g_s_pop_com = g_s_pop[which(!is.na(g_s_pop))]
g_s_country_com = g_s_country[which(!is.na(g_s_pop))]
pop_permanova = adonis(bc_comp ~ g_s_pop_com / g_s_country_com + g_s_load_com, strata = g_s_pop_com )
country_permanova = adonis(bc_comp ~ g_s_load_com + g_s_country_com)
```







##Oomycetes

```{r, echo=FALSE}
concat_name="_oomycete.csv"

g_s=read.csv(paste("~/Dropbox/controlled_metagenomics/results_figures/sweden_germany_combined", concat_name, sep=''), header = TRUE, row.names = 1, stringsAsFactors = FALSE)

g_s_tot = process_gs(g_s, subset=FALSE)

g_s_otus_filter = g_s_tot[[1]]

g_s_country = g_s_tot[[2]]

g_s_pop = g_s_tot[[3]]

g_s_load = g_s_tot[[4]]

bray_curtis <-as.matrix(vegdist(g_s_otus_filter, method="bray", binary=FALSE, diag=FALSE, upper=FALSE, na.rm = FALSE))

#PCoA on bray-curtis
bc_pcoa=pcoa(bray_curtis, correction = "lingoes")

perc_explained=bc_pcoa$values[,3]

####Load vs PCs continued
load_pcoa <- ggplot(data=data.frame(bc_pcoa$vectors), aes(x=Axis.1, y=Axis.2)) + 
  geom_point(aes(color=log10(g_s_load)), cex = 2) +
  scale_color_gradient2(name = "", low = "blue", mid = "white", high = "red") +
  theme_bw() + theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank(), text = element_text(size = 20)) +
  xlab("") + 
  theme(legend.key.width=unit(1,"cm"), axis.title.y = element_blank(), legend.text.align = 1)
# ylab(paste(paste("Axis 2:", (100*round(perc_explained[2],3)), sep=" "), "%", sep=""))

pop_pcoa <-ggplot(data=data.frame(bc_pcoa$vectors), aes(x=Axis.1, y=Axis.2)) + 
  geom_point(aes(color=g_s_country), cex = 2) +
  scale_color_discrete(name="Country") +
  theme_bw() + theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank(), text = element_text(size = 20)) +
  xlab(paste(paste("Axis 1:", (100*round(perc_explained[1],3)), sep=" "), "%", sep="")) +
 theme(legend.key.width=unit(1,"cm"), axis.title.y = element_blank()) 
#  ylab(paste(paste("Axis 2:", (100*round(perc_explained[2],3)), sep=" "), "%", sep=""))

gA <- ggplotGrob(load_pcoa)
gB <- ggplotGrob(pop_pcoa)
gA$widths <- gB$widths
gA <- ggplotGrob(load_pcoa)
gB <- ggplotGrob(pop_pcoa)
gA$widths <- gB$widths

grid.arrange(gA, gB, nrow = 2, left = paste(paste("Axis 2:", (100*round(perc_explained[2],3)), sep=" "), "%", sep=""))

pdf(paste(paste("~/Dropbox/controlled_metagenomics/results_figures/pcoa_bray_curtis_", concat_name, sep=""), ".pdf", sep=""), family = "ArialMT", useDingbats = FALSE)
grid.arrange(gA, gB, nrow = 2, left = paste(paste("Axis 2:", (100*round(perc_explained[2],3)), sep=" "), "%", sep=""))
dev.off()

```

###Which microbe has the highest load across samples?
```{r}
find_top_abundance(g_s_otus_filter)

```

###Do PC1 and PC2 distinguish between samples from germany vs sweden?
First PC1:
```{r, echo = FALSE}
pc1<-bc_pcoa$vectors[,1]
german_pc1<-pc1[g_s_country=="germany"]
sweden_pc1<-pc1[g_s_country=="Sweden"]
wilcox.test(german_pc1, sweden_pc1)
#p=0.0033 W=5367
```

###Next PC2:
```{r, echo = FALSE}
pc2<-bc_pcoa$vectors[,2]
german_pc2<-pc2[g_s_country=="germany"]
sweden_pc2<-pc2[g_s_country=="Sweden"]
wilcox.test(german_pc2, sweden_pc2)
#p=0.0309 W=5694
```

###What is the relationship between load and PC1?
```{r, echo = FALSE}
cor.test(g_s_load, pc1)
```


#PERMANOVA WITH BRAY-CURTIS DISTANCES
Here is some advice on nesting:
https://stats.stackexchange.com/questions/188519/adonis-in-vegan-order-of-variables-or-use-of-strata

```{r}
#exclude NA
bc_comp = bray_curtis[which(!is.na(g_s_pop)), which(!is.na(g_s_pop))]
g_s_load_com = g_s_load[which(!is.na(g_s_pop))]
g_s_pop_com = g_s_pop[which(!is.na(g_s_pop))]
g_s_country_com = g_s_country[which(!is.na(g_s_pop))]
pop_permanova = adonis(bc_comp ~ g_s_pop_com / g_s_country_com + g_s_load_com, strata = g_s_pop_com )
country_permanova = adonis(bc_comp ~ g_s_load_com + g_s_country_com)
```























###How is load associated with Shannon Diversity
#```{r, echo=FALSE}
shannon_div <- vegan::diversity(g_s_otus_filter, index = "shannon")
simpson_div <- vegan::diversity(g_s_otus_filter, index="simpson")
J <- shannon_div/log(specnumber(g_s_otus_filter))
max_tax=apply(g_s_otus_filter, 1, max)
max_tax_names = colnames(g_s_otus_filter)[apply(g_s_otus_filter, 1, which.max)]
div_tog=bind_cols(load=g_s_load, shannon=shannon_div, simpson=simpson_div, J=J, max_tax=max_tax)

J_plot<-ggplot(data=div_tog, aes(x=log10(load), y=J)) + 
  geom_point(pch=20) + 
  geom_smooth(method = "glm", 
    method.args = list(family = quasibinomial(link = 'logit')), color='red', level=0) +
  theme_bw() + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  xlab("Load (Total Microb. Cov./Plant Cov.)") +
  ylab("Evenness (Pielou's J)")

Shannon_div_plot<-ggplot(data=div_tog, aes(x=log10(load), y=shannon)) + 
  geom_point(pch=20) + 
  geom_smooth(method='glm', method.args = list(family = quasibinomial(link = 'logit')), color='red', level=0) +
  theme_bw() + 
  theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  xlab(expression(paste(' Load (log'[10],"(Total Microb. Cov./Plant Cov.))", sep=""))) +
  ylab("Shannon Index (H')")

max_plot<-ggplot(data=div_tog, aes(x=log10(load), y=max_tax)) + 
  geom_point(pch=20) + 
  geom_smooth(method = "glm", 
    method.args = list(family = quasibinomial(link = 'logit')), color='red', level=0) +
  theme_bw() + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  xlab(expression(paste(' Load (log'[10],"(Total Microb. Cov./Plant Cov.))", sep=""))) +
  ylab("Maximum Abundance (% of total)")  + 
  scale_y_continuous(labels = function(x) x * 100) +
  geom_text(aes(label=ifelse(load>0.05,as.character(max_tax_names),'')),position=position_jitter(width=.3,height=.3))

plot_grid(J_plot, max_plot, ncol=1)

pdf(paste(paste("~/Dropbox/controlled_metagenomics/results_figures/div_load", concat_name, sep=""),".pdf",sep=""), family = "ArialMT")
plot_grid(J_plot, max_plot, ncol=1)
dev.off()
```

###Does load affect the overall evenness of the sample?
#```{r, echo = FALSE}
cor.test(g_s_load, J)

cor.test(g_s_load, shannon_div)
```
